src/Interpreters/executeQuery.cpp

```diff
@@ -317,6 +317,10 @@ QueryLogElement logQueryStart(
 
     bool log_queries = settings.log_queries && !internal;
 
+    auto query_log = context->getQueryLog();
```

Comment:
If we don't have a query log at all, why do we even run the code in l. 298 - 318. We could early-out already at the very beginning of `logQueryStart`.

Reply:
No problem to have an extra check.